### PR TITLE
Initial architecture page.

### PR DESCRIPTION
**Pull request by alanconway**
* High-level overview of the cluster logging architecture.
* Summary of work in progress on the design of future enhancments.
* Links to further details.

This is _not_ intended to duplicate information from JIRAs, enhancment requests, user documentation, or the source. It is intended to make it easier to understand the project as a
whole and to provide links to help find the details you need.

You can view the .adoc page directly on github (ugly), or a prettier [GitHub pages version](https://alanconway.github.io/cluster-logging-operator/architecture/)

### CODE REVIEW

File: docs/architecture.adoc [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+=== Operators and APIs
+
+[.thumb]
+.Operators and APIs
+image::overview.svg[]
+
+=== Collection and forwarding
+
+[.thumb]
+.Collection and forwarding
```
Comment (vimalk78) [inline docs/architecture.adoc:48]: Add a section to show what type of logs define which input app/infra/audit.

Comment (alanconway) [reply docs/architecture.adoc:70]: Done, also elaborated on the overview pictures if you want to check again.

**Comment by alanconway** [comment]
Can't reopen, new PR: https://github.com/openshift/cluster-logging-operator/pull/707

**Pull request closed without merging**